.github/ISSUE_TEMPLATE/work-in-progress.md

```diff
@@ -0,0 +1,32 @@
+---
+name: Work in progress
+about: Describe a new feature that you are currently working on
+title: ''
+labels: work-in-progress
+assignees: ''
+
+---
+
+**Abstract**
+
+A brief description of the work being done.
+
+**Motivation**
+
+Describe the need for the work being done:
+
+* What problem is it trying to solve?
+* Who is affected by the change?
```

Comment:
Would an entry for "Files impacted" be worthwhile?

Reply:
We could ask for files / classes impacted, but that feels like more detail than necessary. Once people have modifications underway, I'd like them to link to the in-progress branch where they're working.

Reply:
Yes, good point.  My only thought here was that it would give a way to communicate to others if the "ground beneath their feet" was about to change. i.e. if there is some major change that is going to affect files where they are currently working on changes.  But agree that it might be a bit much to ask for...